server/src/main/java/com/paypal/selion/grid/SeLionGridLauncher.java

```diff
@@ -108,43 +196,35 @@ public void boot(String[] args) throws Exception {
             ConfigParser.setConfigFile(helper.getParamValue(SeLionGridConstants.SELION_CONFIG_ARG));
         }
 
-        try {
-            GridRole role = GridRole.find(args);
+        if (launchers.containsKey(role)) {
             logEnvironment();
 
-            switch (role) {
-            case NOT_GRID:
-                LOGGER.info("Launching a standalone server");
-                final RemoteControlConfiguration configuration = RemoteControlLauncher.parseLauncherOptions(args);
-                final SeleniumServer proxy = new SeleniumServer(configuration);
-                type = proxy;
-                proxy.boot();
-                break;
-            case HUB:
-                LOGGER.info("Launching a selenium grid server");
-                final GridHubConfiguration ghc = GridHubConfiguration.build(args);
-                final Hub h = new Hub(ghc);
-                type = h;
-                h.start();
-                break;
-            case NODE:
-                LOGGER.info("Launching a selenium grid node");
-                final RegistrationRequest rr = RegistrationRequest.build(args);
-                final SelfRegisteringRemote remote = new SelfRegisteringRemote(rr);
-                type = remote;
-                remote.startRemoteServer();
-                remote.startRegistrationProcess();
-                break;
-            default:
-                throw new RuntimeException("NI");
+            try {
+                launchers.get(role).launch(args, LOGGER);
+                type = launchers.get(role).getType();
+            } catch (Exception e) {
+                launchers.get(role).printUsage();
+                e.printStackTrace();
             }
-        } catch (GridConfigurationException e) {
-            GridDocHelper.printHubHelp(e.getMessage());
-            e.printStackTrace();
-            throw e;
-        } catch (RuntimeException rex) {
-            rex.printStackTrace();
-            throw rex;
+        } else {
+            throw new GridConfigurationException("Unknown role: " + role);
+        }
+    }
+
+    private static void printInfoAboutRoles(CommandLineOptionHelper helper) {
+        if (helper.hasParamValue("-role")) {
+            CliUtils.printWrappedLine("", "Error: the role '" + helper.getParamValue("-role")
+                    + "' does not match a recognized server role\n");
+        } else {
+            CliUtils.printWrappedLine("",
+                    "Error: -role option needs to be followed by the value that defines role of this component in the grid\n");
         }
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [System.out.println is used](https://www.codacy.com/app/SeLion/paypal-SeLion/file/2120612344/issues/source?bid=2928425&fileBranchId=3146721#l222)